presentation/presentation.go

```diff
@@ -0,0 +1,210 @@
+		numRedo := strconv.FormatInt(int64(rs.NumRedo), 10)
+		loc := rs.Location.String()
+		line = append(line, index, timeNsStr, numEval, numRedo, loc)
+		tableProfile.Append(line)
+	}
+	if tableProfile.NumLines() > 0 {
+		fmt.Println("\nProfile:")
+		tableProfile.Render()
+	}
+}
+
+func checkStrLimit(input string, limit int) string {
+	if limit > 0 && len(input) > limit {
+		input = input[:limit] + "..."
+		return input
+	}
+	return input
+}
+
+func generateTableBindings(writer io.Writer, keys []ResultKey, result EvalResult, prettyLimit int) *tablewriter.Table {
+	table := tablewriter.NewWriter(writer)
+	table.SetAlignment(tablewriter.ALIGN_CENTER)
+	table.SetAutoFormatHeaders(false)
+	header := make([]string, len(keys))
+	for i := range header {
+		header[i] = keys[i].string()
+	}
+	table.SetHeader(header)
+
+	for _, row := range result.Result {
+		printPrettyRow(table, keys, row, prettyLimit)
+	}
+	return table
+}
+
+func printPrettyRow(table *tablewriter.Table, keys []ResultKey, result rego.Result, prettyLimit int) {
+	buf := []string{}
+	for _, k := range keys {
+		v, ok := k.selectVarValue(result)
+		if ok {
+			js, err := json.Marshal(v)
+			if err != nil {
+				buf = append(buf, err.Error())
+			} else {
+				s := checkStrLimit(string(js), prettyLimit)
+				buf = append(buf, s)
+			}
+		}
+	}
+	table.Append(buf)
+}
+
+func generateTableMetrics(writer io.Writer) *tablewriter.Table {
+	table := tablewriter.NewWriter(writer)
+	table.SetHeader([]string{"Num", "Name", "Value"})
+	table.SetAlignment(tablewriter.ALIGN_CENTER)
+	table.SetColumnAlignment([]int{tablewriter.ALIGN_RIGHT, tablewriter.ALIGN_LEFT,
+		tablewriter.ALIGN_LEFT})
+	table.SetHeaderColor(
+		tablewriter.Colors{tablewriter.FgMagentaColor},
+		tablewriter.Colors{tablewriter.FgMagentaColor},
+		tablewriter.Colors{tablewriter.FgMagentaColor},
+	)
+	return table
+}
+
+func generateTableProfile(writer io.Writer) *tablewriter.Table {
+	table := tablewriter.NewWriter(writer)
+	table.SetHeader([]string{"Num", "Time", "Num Eval", "Num Redo", "Location"})
+	table.SetAlignment(tablewriter.ALIGN_CENTER)
+	table.SetColumnAlignment([]int{tablewriter.ALIGN_RIGHT, tablewriter.ALIGN_LEFT,
+		tablewriter.ALIGN_LEFT, tablewriter.ALIGN_LEFT,
+		tablewriter.ALIGN_LEFT})
+	table.SetHeaderColor(
+		tablewriter.Colors{tablewriter.FgMagentaColor},
+		tablewriter.Colors{tablewriter.FgMagentaColor},
+		tablewriter.Colors{tablewriter.FgMagentaColor},
+		tablewriter.Colors{tablewriter.FgMagentaColor},
+		tablewriter.Colors{tablewriter.FgMagentaColor},
+	)
+	return table
+}
+
+func populateTableMetrics(data map[string]interface{}, table *tablewriter.Table, prettyLimit int) {
+	var idx int64 = 1
+	for varName, varValueInterface := range data {
+		line := []string{}
+		index := strconv.FormatInt(idx, 10)
+		varValue := checkStrLimit(fmt.Sprintf("%v", varValueInterface), prettyLimit)
+		line = append(line, index, varName, varValue)
+		table.Append(line)
+		idx++
+	}
+}
+
+// ResultKey holds variable information
```

Comment:
Can we avoid exporting this struct and move the logic that constructs the result keys into this package? 

Currently this logic is duplicated in the repl and eval packages.

Reply:
Un-exported `resultKey` struct